cosmwasm/enclaves/shared/contract-engine/src/contract_validation.rs

```diff
@@ -351,100 +357,123 @@ pub fn validate_basic_msg(
 }
 
 /// Verify all the parameters sent to the enclave match up, and were signed by the right account.
-#[allow(unused_variables)]
+#[allow(clippy::too_many_arguments)]
 pub fn verify_params(
     sig_info: &SigInfo,
     sent_funds: &[Coin],
     sender: &CanonicalAddr,
     contract_address: &HumanAddr,
     msg: &SecretMessage,
-    og_msg: &[u8],
+    #[cfg(feature = "light-client-validation")] og_msg: &[u8],
+    should_validate_sig_info: bool,
+    should_validate_input: bool,
+    handle_type: HandleType,
```

Comment:
Split this up into multiple functions... at this point it's too complex. We shouldn't just ignore clippy warns and keep adding complexity

Reply:
The best I can do here is this - https://github.com/scrtlabs/SecretNetwork/commit/d91cb2179, otherwise it's just going to be inlining both of these ifs in `init()` & `handle()` which doesn't feel right.